openslides/core/static/js/core/core.js

```diff
@@ -592,33 +628,208 @@ angular.module('OpenSlidesApp.core.site', [
+            data[countdown.uuid] = {
+                "status": "stop",
+                "countdown_time": newDuration
+            };
+            $http.post('/rest/core/projector/1/update_elements/', data);
+        };
+        $scope.resetCountdown = function (countdown) {
+            var data = {};
+            data[countdown.uuid] = {
+                "status": "stop",
+                "countdown_time": countdown.default,
+            };
+            $http.post('/rest/core/projector/1/update_elements/', data);
+        };
+
+        // *** message functions ***
+        $scope.addMessage = function () {
+            $http.post('/rest/core/projector/1/activate_elements/', [{
+                    name: 'core/message',
+                    visible: false,
+                    index: $scope.messages.length,
+                    message: '',
+                    stable: true
+            }]);
+        };
+        $scope.removeMessage = function (message) {
+            $http.post('/rest/core/projector/1/deactivate_elements/', [message.uuid]);
+        };
+        $scope.showMessage = function (message) {
+            var data = {};
+            // if current message is activated, deactivate all other messages
+            if ( !message.visible ) {
+                for (var i=0; i<$scope.messages.length; i++) {
+                    if ( $scope.messages[i].uuid == message.uuid ) {
+                        data[$scope.messages[i].uuid] = { "visible": true };
+                    } else {
+                        data[$scope.messages[i].uuid] = { "visible": false };
+                    }
+                }
+            } else {
+                data[message.uuid] = { "visible": false };
+            }
+            $http.post('/rest/core/projector/1/update_elements/', data);
+        };
+        $scope.editMessage = function (message) {
+            var data = {};
+            data[message.uuid] = {
+                "message": message.message,
+            };
+            $http.post('/rest/core/projector/1/update_elements/', data);
+            message.editMessageFlag = false;
+        };
+
+        // *** projector controls ***
+        $scope.scrollLevel = Projector.get(1).scroll;
+        $scope.scaleLevel = Projector.get(1).scale;
+        $scope.controlProjector = function (action, direction) {
+            $http.post('/rest/core/projector/1/control_view/', {"action": action, "direction": direction});
+        };
+        $scope.editCurrentSlide = function () {
+            $.each(Projector.get(1).elements, function(key, value) {
+                if (value.name != 'core/clock' &&
+                    value.name != 'core/countdown' &&
+                    value.name != 'core/message' ) {
+                    $state.go(value.name.replace('/', '.')+'.detail.update', {id: value.id });
+                }
+            });
+        };
+    }
+])
```

Comment:
I do not know whether we need to cancel the intervals on destruction of this controller too.
@ostcar What happens if the user is here (projector tab) and clicks on e. g. Motion. Is this controller destroyed and rebuild if here comes back later?
If yes @emanuelschuetze has to implement a listener on the destruction event to cancel the intervals so the browser memory keeps clean.